code/datums/drift_handler.dm

```diff
@@ -0,0 +1,251 @@
+///Component that handles drifting
+///Manages a movement loop that actually does the legwork of moving someone
+///Alongside dealing with the post movement input blocking required to make things look nice
+/datum/drift_handler
+	var/atom/movable/parent
+	var/atom/inertia_last_loc
+	var/old_dir
+	var/datum/move_loop/smooth_move/drifting_loop
+	///Should we ignore the next glide rate input we get?
+	///This is to some extent a hack around the order of operations
+	///Around COMSIG_MOVELOOP_POSTPROCESS. I'm sorry lad
+	var/ignore_next_glide = FALSE
+	///Have we been delayed? IE: active, but not working right this second?
+	var/delayed = FALSE
+	var/block_inputs_until
+	/// How much force is behind this drift.
+	var/drift_force = 1
+
+/// Accepts three args. The direction to drift in, if the drift is instant or not, and if it's not instant, the delay on the start
+/datum/drift_handler/New(atom/movable/parent, inertia_angle, instant = FALSE, start_delay = 0, drift_force = 1)
+	. = ..()
+	src.parent = parent
+	var/flags = MOVEMENT_LOOP_OUTSIDE_CONTROL
+	if(instant)
+		flags |= MOVEMENT_LOOP_START_FAST
+	src.drift_force = drift_force
+	drifting_loop = GLOB.move_manager.smooth_move(moving = parent, angle = inertia_angle, delay = 1, subsystem = SSspacedrift, priority = MOVEMENT_SPACE_PRIORITY, flags = flags)
+
+	if(!drifting_loop) //Really want to qdel here but can't
+		return COMPONENT_INCOMPATIBLE
```

Comment:
pointless return value in New from when this was a component, you also don't check anything after creating these datums ?